File: EventVenueParkingSystem.API/wwwroot/js/customer/payment.js
async function pay() {


    let id =
        Number(
            document.getElementById("bookingId").value
        );



    let data = {

        paymentMethod: "Simulated"

    };



    try {


        let result =
            await apiCall(
                "/payments/booking/" + id,
                "POST",
                data
            );



        document.getElementById("msg")
            .innerHTML =
            "Payment Successful";


    }

    catch (error) {

        document.getElementById("msg")
            .innerHTML =
            "Payment Failed";

    }


}

File: EventVenueParkingSystem.API/Models/Payment.cs
namespace EventParkingReservationSystem.API.Models
{
    public class Payment
    {
        public int PaymentId { get; set; }

        // =====================================================
        // Booking
        // One successful payment per booking
        // =====================================================
        public int BookingId { get; set; }

        public Booking? Booking { get; set; }


        // =====================================================
        // Payment Amount
        // Booking TotalAmount snapshot
        // =====================================================
        public decimal Amount { get; set; }


        // =====================================================
        // Payment Method
        // Card / Cash / Simulated
        // =====================================================
        public string PaymentMethod { get; set; } = "Simulated";


        // =====================================================
        // Payment Status
        // Pending / Successful / Failed
        // =====================================================
        public string Status { get; set; } = "Pending";


        // =====================================================
        // Transaction Reference
        // Example: PAY-2026-000001
        // =====================================================
        public string TransactionReference { get; set; } = string.Empty;


        // =====================================================
        // Payment Date
        // =====================================================
        public DateTime? PaidAt { get; set; }


        // =====================================================
        // Audit
        // =====================================================
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }
    }
}